spec/unit/network/http/api/v3/indirected_routes_spec.rb

```diff
@@ -198,7 +207,12 @@ def a_request_that_searches(key, request = {})
     end
 
     it "should include the environment in the query string of the URI" do
-      handler.class.request_to_uri(request).should == "#{url_prefix}/foo/with%20spaces?environment=myenv&foo=bar"
+      handler.class.request_to_uri(request).should == "#{master_url_prefix}/foo/with%20spaces?environment=myenv&foo=bar"
+    end
+
+    it "should include the correct url prefix if it is a ca request" do
+      request.stubs(:indirection_name).returns("certificate")
+      handler.class.request_to_uri(request).should == "#{ca_url_prefix}/certificate/with%20spaces?environment=myenv&foo=bar"
```

Comment:
Requests originating from the agent will include the environment query parameter, because all agent URLs are constructed via the indirector.  For the Rack/Webrick masters, the requests will fail if this parameter is not provided.

For the Puppet Server CA endpoints, I suspect that it would be acceptable to ignore this query parameter rather than returning a failure response, but that is a question for product, and is orthogonal to this PR.

We should not expect the agent to stop sending this parameter until there is a significant rework of the client-side code that involves removing the indirector from the process of constructing the URLs.